[docs] Fix Linking.createURL docs for Expo Go (#25018)
expo/expo: docs/pages/guides/linking.mdx

To save you the trouble of inserting a bunch of conditionals based on the environment that you're in and hardcoding urls, we provide some helper methods in our extension of the `Linking` module. When you want to provide a service with a url that it needs to redirect back into your app, you can call `Linking.createURL()` and it will resolve to the following:

- _Builds and development builds_: `myapp://` (see [#linking-to-your-app](#linking-to-your-app))
- _Development in Expo Go_: `exp://127.0.0.1:8081`. For SDK 48 and lower, the port number is `19000`.

> When loading published updates in Expo Go, the URL that is created in `createURL` depdends on the manner in which the project was launched and shouldn't be relied upon to be consistent or stable. For applications that require a stable URL (authorization provider redirects for example), a development build and custom scheme should be used (see [#linking-to-your-app](#linking-to-your-app)).

Reviewer: ```suggestion
When loading published updates in Expo Go, the URL that is created by `Linking.createURL()` depends on the manner in which the project was launched and shouldn't be relied upon to be consistent or stable. For applications that require a stable URL (authorization provider redirects, for example), you should use a development build with a custom scheme instead (see [#linking-to-your-app](#linking-to-your-app)).
```

> - _App loaded via QR code or via menus in Expo Go_: `exp://u.expo.dev/[project-id]/group/[update-group-id]` or `exp://u.expo.dev/[project-id]/update/[update-id]`. The project ID is stable but the update group ID and update ID are not.

Reviewer: ```suggestion
- _Project loaded via QR code or via menus in Expo Go_: `exp://u.expo.dev/[project-id]/group/[update-group-id]` or `exp://u.expo.dev/[project-id]/update/[update-id]`. The project ID is stable but the update group ID and update ID are not.
```

This will resolve into the following, depending on the environment:

> - _Builds and development builds_: `myapp://path/into/app?hello=world`

Reviewer: ```suggestion
- _Production and development builds_: `myapp://path/into/app?hello=world`
```

- _Development in Expo Go_: `exp://127.0.0.1:8081/--/path/into/app?hello=world`. For SDK 48 and lower, the port number is `19000`.

> Again, when loading published updates in Expo Go this behavior is undefined, but may look something like the following:

Reviewer: The section above defines the behavior. It would be more accurate to say here that the update group ID varies from update to update.
